File: solutions/209.minimum-size-subarray-sum.java
/*
 * @lc app=leetcode id=209 lang=java
 *
 * [209] Minimum Size Subarray Sum
 */

// @lc code=start
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int n = nums.length;
        int i=0, j=0;
        int sum = 0;
        int ans = n+1;
        while(j<n) {
            sum+=nums[j];
            while(sum >= target) {
                ans = Math.min(ans, j-i+1);
                sum -= nums[i++];
            }             
            ++j;
        }
        return ans = ans==n+1 ? 0 : ans;
    }
}
// @lc code=end

File: solutions/209.minimum-size-subarray-sum.rb
# @param {Integer} target
# @param {Integer[]} nums
# @return {Integer}
def min_sub_array_len(target, nums)
  i=0
  sum = 0
  ans = nums.size+1
  (0...nums.size).each do |j|
    sum += nums[j]
    while sum >= target
      ans = [ans, j-i+1].min
      sum -= nums[i]
      i += 1
    end
  end
  ans == nums.size+1 ? 0 : ans
end
